content/docs/start/experiments-trail/experiments-1.md

```diff
@@ -0,0 +1,334 @@
+                -p model.conv_units \
+                -p train.epochs \
+                -d data/images \
+                -m metrics.json \
+                python3 src/train.py
+```
+
+The command tells DVC to create an experiment named `train`, and that for any
+change in `data/images/`, `model.conv_units` or `train.epochs`, we run an
+experiment using `src/train.py` that produces a new `metrics.json` file.
+
+DVC creates `dvc.yaml` file and modifies `.gitignore` for Git to ignore certain
+artifacts.
+
+```dvc
+$ git add dvc.yaml .gitignore
+$ git commit -m "added data and the experiment"
+```
+
+DVC is ready to run the experiments now!
+
+</details>
+
+## Running the experiment with default parameters
+
+The purpose of `dvc exp` subcommands is to run the pipeline for ephemeral
+experiments. By _ephemeral_ we mean the experiments can be run without
+committing parameter and dependency changes to Git. Instead the artifacts
+produced for each experiment are tracked by DVC and persisted on demand.
+
+Running the experiment with default hyperparameter values requires only the
+command:
+
+```dvc
+$ dvc exp run
+...
+Reproduced experiment(s): exp-7683f
+Experiment results have been applied to your workspace.
+
+To promote an experiment to a Git branch run:
+
+        dvc exp branch <exp>
+```
+
+It runs the pipeline starting from the basic dependencies and produces
+`metrics.json` file for the default state.
+
+<details>
+
+### If you used `dvc repro` before
+
+Earlier versions of DVC uses `dvc repro` to run the pipeline. If you already
+have a DVC project, you may already be using `dvc repro`.
+
+In DVC 2.0 `dvc exp run` supersedes `dvc repro`. Both of these commands run the
+pipeline.
+
+We use `dvc repro` to run the pipeline as found in the <abbr>workspace</abbr>.
+All the parameters and dependencies are retrieved from the current workspace. It
+doesn't use any special objects to track the experiments. When you have large
+number of experiments that you don't want to commit all to Git, it's better to
+use `dvc exp run`. It allows to change the parameters quickly, can track the
+history of artifacts and has facilities to compare these experiments easily.
+
+`dvc repro` is still available to run the pipeline when these extra features are
+not needed.
+
+</details>
+
+## Running the experiment by setting parameters
+
+Now let's do some more experimentation.
+
+DVC allows to update the parameters defined in the pipeline without modifying
+the files manually. We use this feature to set the convolutional units in
+`train.py`.
+
+```dvc
+$ dvc exp run --set-param model.conv_units=24
+...
+Reproduced experiment(s): exp-6c06d
+Experiment results have been applied to your workspace.
+
+To promote an experiment to a Git branch run:
+
+        dvc exp branch <exp>
+```
+
+When you run `dvc exp run` with `--set-param`, it updates the parameter file. We
+can see the effect of it by looking at the diff.
+
+```dvc
+$ git diff params.yaml
```

Comment:
Why not introduce `exp diff` instead? Assuming it works with a single experiment vs. `HEAD` version. Otherwise maybe nothing: not sure seeing this diff is necessary.

Reply:
`exp diff` reports the results in _metrics_, we are telling how parameters are updated here.

Reply:
Ok. Well `exp diff` shows both params and metrics. If you only want the metrics there's `params diff` too 🔀

Reply:
I mean the params lol

Reply:
I want to show how the file has changed with `dvc exp run -S`, with an external tool. I could use the plain `diff` by writing the params file somewhere before the experiment but this is simpler IMO.